sqlite_query.c

```diff
@@ -104,24 +302,111 @@ sqlite_convert_to_pg(Oid pgtyp, int pgtypmod, sqlite3_stmt * stmt, int stmt_coli
+					case SQLITE_FLOAT:
+						{
+							double		value = sqlite3_column_double(stmt, stmt_colid);
+							Datum		d = DirectFunctionCall1(float8_timestamptz, Float8GetDatum((float8) value));
+							return (struct NullableDatum) {d, false};
+						}
+					case SQLITE_BLOB:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+							break;
+						}
+					case SQLITE3_TEXT:
+					{
+						if (value_byte_size_blob_or_utf8)
+							valstr = sqlite_text_value_to_pg_db_encoding(stmt, stmt_colid);
+						else
+							elog(ERROR, "Void text disallowed for PostgreSQL float4 column");
+						break;
+					}
+					default:
+					{
+						sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+						break;
+					}
 				}
 				break;
 			}
 		case NUMERICOID:
 			{
-				double		value = sqlite3_column_double(stmt, stmt_colid);
-
-				valstr = DatumGetCString(DirectFunctionCall1(float8out, Float8GetDatum((float8) value)));
+				switch (sqlite_value_affinity)
+				{
+					case SQLITE_INTEGER:
+					case SQLITE_FLOAT: /* <-- proper and recommended SQLite affinity of value for pgtyp */
+						{
+							double		value = sqlite3_column_double(stmt, stmt_colid);
+							valstr = DatumGetCString(DirectFunctionCall1(float8out, Float8GetDatum((float8) value)));
+							break; /* !!! use valstr later! */
+						}
+					case SQLITE_BLOB:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+							break;
+						}
+					case SQLITE3_TEXT:
+					{
+						if (value_byte_size_blob_or_utf8)
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+						else
+							elog(ERROR, "Void text disallowed for PostgreSQL numeric column");
+						break;
+					}
+					default:
+					{
+						sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+						break;
+					}
+				}
+				break;
+			}
+			case UUIDOID:
+				{
+				switch (sqlite_value_affinity)
+				{
+					case SQLITE_INTEGER:
+					case SQLITE_FLOAT:
+						{
+							sqlite_value_to_pg_error (pgtyp, pgtypmod, stmt, attnum, sqlite_value_affinity, affinity_for_pg_column, value_byte_size_blob_or_utf8);
+							break;
+						}
+					case SQLITE_BLOB: /* <-- first proper and recommended SQLite affinity of value for pgtyp */
+						{
+							if (value_byte_size_blob_or_utf8 != UUID_LEN)
+							{
+								ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
+												errmsg("PostgreSQL uuid data type allows only %d bytes SQLite blob value", UUID_LEN),
+												errhint("incorrect value is %d bytes length", value_byte_size_blob_or_utf8)));
+								break;
+							}
+						else
```

Comment:
Please check the indent.

Reply:
What is conventional indent in this case? I have seen such `errmsg` and `errhint` indent as
```
errmsg
errhint
```
Please give me preferred example for `ereport`.

Reply:
Fixed. There was 2 tabs.

Reply:
Thank for updating. Confirmed.